e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -70,9 +70,24 @@ private void assertThatAccessStructuredWasFetched(String conversationId, Documen
         assertThat(accessStructured.get("taskId")).isNotNull();
     }
 
-    private void assertThatAccessDocumentWasFetched(String documentId, List<Document> accessDocuments) {
-        Document document = accessDocuments.get(0);
-        assertThat(document.get("objectName")).isEqualTo(documentId + ".json");
+    private Document theDocumentTaskUpdatesTheRecord(String conversationId) {
+        var gpcAccessDocument = (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT);
+        return getFirstDocumentIfItHasObjectNameOrElseNull(gpcAccessDocument);
+    }
+
+    private Document getFirstDocumentIfItHasObjectNameOrElseNull(Document gpcAccessDocument) {
+        var documentList = gpcAccessDocument.get("documents", Collections.emptyList());
+        if(!documentList.isEmpty()) {
+            Document document = (Document) documentList.get(0);
+            if (document.get("objectName") != null) {
+                return document;
+            }
+        }
+        return null;
```

Comment:
I am not a fan of returning nulls. Could use "Optional" instead?

Reply:
Mongo BSON library uses null to signal that something doesn't exist. Leaving as-is to be consistent